File: src/main/java/com/yly/jdk8/StreamTest.java
package com.yly.jdk8;

import com.google.common.collect.Lists;

import java.util.ArrayList;
import java.util.List;
import java.util.function.Function;
import java.util.stream.Collectors;
import java.util.stream.Stream;

/**
 * @author yiliyang
 * @version 1.0
 * @date 2021/6/22 上午10:15
 * @since 1.0
 */
public class StreamTest {

    public static void main(String[] args) {
        Stream<Integer> stream = Stream.of(1, 2, 3);
        Stream<Integer> another = Stream.of(4, 5, 6);
        Stream<Integer> third = Stream.of(7, 8, 9);
        Stream<Integer> more = Stream.of(0);
        List result = Stream.of(stream, another, third, more).flatMap(integerStream -> integerStream.map(integer -> integer * 10)).
                collect(Collectors.toList());
        System.out.println(result);
    }
}

File: src/main/java/com/yly/StreamTest.kt
package com.yly

import java.time.*
import java.time.format.DateTimeFormatter
import java.time.temporal.ChronoField
import java.util.*
import java.util.function.BiConsumer
import java.util.function.Consumer
import java.util.function.Predicate
import java.util.function.Supplier
import java.util.stream.Collectors
import java.util.stream.LongStream
import kotlin.reflect.jvm.internal.impl.load.kotlin.JvmType

/**
 * @author    yiliyang
 * @date      2020/12/3 下午2:39
 * @version   1.0
 * @since     1.0
 */
fun main() {

    val list = listOf(1, 2, 3)

    println(
        list.stream().parallel().collect<MutableList<Int>>(
            ::ArrayList,
            { t, u -> t.add(u) },
            { t, u -> t.addAll(u) },
        )
    )
//
//    label@ for (i in 0..2) {
//        if (i == 1) {
//            continue@label
//        }
//        println(i)
//    }

//    val formater = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
//    println(formater.format(LocalDate.now()))
}